modules/exoplanets/rendering/renderableorbitdisc.cpp

```diff
@@ -0,0 +1,325 @@
+        glm::scale(glm::dmat4(1.0), glm::dvec3(data.modelTransform.scale));
+
+    glm::dmat4 modelViewTransform = data.camera.combinedViewMatrix() * modelTransform;
+
+    _shader->setUniform(
+        _uniformCache.modelViewProjection,
+        data.camera.projectionMatrix() * glm::mat4(modelViewTransform)
+    );
+    _shader->setUniform(_uniformCache.textureOffset, _offset);
+    _shader->setUniform(_uniformCache.opacity, _opacity);
+    _shader->setUniform(_uniformCache.eccentricity, _eccentricity);
+    _shader->setUniform(_uniformCache.semiMajorAxis, _size);
+
+    ghoul::opengl::TextureUnit unit;
+    unit.activate();
+    _texture->bind();
+    _shader->setUniform(_uniformCache.texture, unit);
+
+    glEnablei(GL_BLEND, 0);
+    glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+    glDepthMask(false);
+    glDisable(GL_CULL_FACE);
+
+    glBindVertexArray(_quad);
+    glDrawArrays(GL_TRIANGLES, 0, 6);
+
+    _shader->deactivate();
+
+    // Restores GL State
+    global::renderEngine.openglStateCache().resetBlendState();
+    global::renderEngine.openglStateCache().resetDepthState();
+    global::renderEngine.openglStateCache().resetPolygonAndClippingState();
+}
+
+void RenderableOrbitDisc::update(const UpdateData& data) {
+    if (_shader->isDirty()) {
+        _shader->rebuildFromFile();
+        _uniformCache.modelViewProjection = _shader->uniformLocation(
+            "modelViewProjectionTransform"
+        );
+        _uniformCache.textureOffset = _shader->uniformLocation("textureOffset");
+        _uniformCache.opacity = _shader->uniformLocation("opacity");
+        _uniformCache.texture = _shader->uniformLocation("discTexture");
+        _uniformCache.eccentricity = _shader->uniformLocation("eccentricity");
+        _uniformCache.semiMajorAxis = _shader->uniformLocation("semiMajorAxis");
+    }
+
+    if (_planeIsDirty) {
+        createPlane();
+        _planeIsDirty = false;
+    }
+
+    if (_textureIsDirty) {
+        loadTexture();
+        _textureIsDirty = false;
+    }
+}
+
+void RenderableOrbitDisc::loadTexture() {
+    if (_texturePath.value() != "") {
+        std::unique_ptr<ghoul::opengl::Texture> texture =
+            ghoul::io::TextureReader::ref().loadTexture(absPath(_texturePath));
+
+        if (texture) {
+            LDEBUGC(
+                "RenderableOrbitDisc",
+                fmt::format("Loaded texture from '{}'", absPath(_texturePath))
+            );
+            _texture = std::move(texture);
+
+            _texture->uploadTexture();
+            _texture->setFilter(ghoul::opengl::Texture::FilterMode::AnisotropicMipMap);
+
+            _textureFile = std::make_unique<ghoul::filesystem::File>(_texturePath);
+            _textureFile->setCallback(
+                [&](const ghoul::filesystem::File&) { _textureIsDirty = true; }
+            );
+        }
+    }
+}
+
+void RenderableOrbitDisc::createPlane() {
+    const GLfloat size = _size.value() * (1.0 + _eccentricity.value());
```

Comment:
This line should work without the `value()`  calls as well as the properties should automatically convert to their underlying type:

```
FloatProperty p(...);
float f = p;

p = 2.f;
```